hugegraph-core/src/main/java/com/baidu/hugegraph/backend/store/BackendEntryIterator.java

```diff
@@ -115,8 +118,17 @@ protected final boolean reachLimit() {
     }
 
     protected final boolean reachLimit(long count) {
-        checkInterrupted();
-        return this.query.reachLimit(count);
+        try {
+            checkInterrupted();
+            return this.query.reachLimit(count);
```

Comment:
it's ok to just catch `checkInterrupted()`, we assume other exception will be handled by upper layer.